### DISCUSSION TITLE
Modal size

### DISCUSSION THREAD

**Discussion by jaz000** [Budibase/budibase#7161] [Help] [answered]
Is it possible to change the size of a modal screen? (I e.what in modal mode)

**Comment by melohagan** [comment, marked as answer]
Hey @jaz000 

It is possible with an embed component: https://docs.budibase.com/docs/custom-css#global-styling

Add the embed component with the following content:
```html
<style>
.actions {
  width: 20% !important;
}
.spectrum-Modal {
  width: 20% !important;
}
iframe {
  height: 400px !important;
}
</style>
```

The result is something like so:

**BEFORE**
![Screenshot 2022-08-09 at 11 29 58](https://user-images.githubusercontent.com/101575380/183626993-d98120bf-fabb-4049-96ec-3b5fad8d2d6d.png)

**AFTER**
![Screenshot 2022-08-09 at 11 29 22](https://user-images.githubusercontent.com/101575380/183626884-50980ec3-a0de-4ed4-a8dd-f898ddabf135.png)

**Reply by JayP718** [reply]
Thanks, this helps a ton.

**Reply by jaz000** [reply]
Super yeah that works - thanks!

**Reply by josh99smith** [reply]
Is there anyway to call different variations of that modal size on the same screen? I have a main view that users can click on different components and they open up different modals. However, I am wondering if its possible to havbe different modal sizes depending on which item they click on the screen?

**Reply by josh99smith** [reply]
> Is there anyway to call different variations of that modal size on the same screen? I have a main view that users can click on different components and they open up different modals. However, I am wondering if its possible to havbe different modal sizes depending on which item they click on the screen?

Was able to acheive this by storing this code as a state called "ModalSize"...then updating that state on button presses and changing the deminsions of the ifram modal to meet my needs for each case.

```
<style>
.actions {
  width: 55% !important;
}
.spectrum-Modal {
  width: 55% !important;
}
iframe {
  height: 150px !important;
}
</style>
```

**Reply by pkondzior** [reply]
Can you elaborate how the ModalSize is being passed through into CSS here ? How the classes are being selected?

**Reply by NicolasFabre81** [reply]
I do not manage to make it work... 
I tried to put the embed in the modal screen itself, or in the root page : no effect.
I'd like to have small modals, like popup messages...

**Reply by JayP718** [reply]
@NicolasFabre81 theres a new action called show alert. Might help for your use case.

**Reply by NicolasFabre81** [reply]
Surely this is what I need ! But I see this is still in develop mode isnt't it ? In which version will it be available ?
https://github.com/Budibase/budibase/pull/7396

**Reply by JayP718** [reply]
> Surely this is what I need ! But I see this is still in develop mode isnt't it ? In which version will it be available ?
> 
> https://github.com/Budibase/budibase/pull/7396

It can actually already be found in the latest update. Be sure to update both Budibase and the individual app.

**Reply by josh99smith** [reply]
> Can you elaborate how the ModalSize is being passed through into CSS here ? How the classes are being selected?

So you will want to create an Embed object on your page first. The value of the embed object should be a new variable/state. I callled mine "ModalSizeData"..

Then, wherever you want to trigger the modal at whatever size you need, you will just create an action that updates the  ModalSizeData state to the CSS code below. And then Navigates to your page with the "open modal" option selected.

`<style>
.actions {
  width: 55% !important;
}
.spectrum-Modal {
  width: 55% !important;
}
iframe {
  height: 150px !important;
}
</style>`

**Reply by NicolasFabre81** [reply]
@JayP718 I'm running V1.2.20 so too old version indeed !
@josh99smith Thanks for the details I'll try

**Reply by NicolasFabre81** [reply]
Hello @melohagan ,
i just pasted your CSS code in a embed tool, and I get this :
<img width="777" alt="image" src="https://user-images.githubusercontent.com/28592069/211845347-81666cf6-7c6a-4e75-9cf8-898c9eaf78c3.png">

How to make the table width fit the modal screen width ? 
(I have no width limit anywhere)

**Reply by josh99smith** [reply]
> Hello @melohagan , i just pasted your CSS code in a embed tool, and I get this : <img alt="image" width="777" src="https://user-images.githubusercontent.com/28592069/211845347-81666cf6-7c6a-4e75-9cf8-898c9eaf78c3.png">
> 
> How to make the table width fit the modal screen width ? (I have no width limit anywhere)

Did you try putting the table into a container element? And then adjusting the width of the container?

**Reply by NicolasFabre81** [reply]
In fact I just had to update the iframe width in the CSS code and it's ok, thank you !

**Reply by kerbersystems** [reply]
For anyone who comes across this thread, I solved the fill issue using the .spectrum-dialog style.

`<style> .actions {   width: 75% !important; } .spectrum-Modal {   width: 75% !important; } iframe {   height: 640px !important;  } .spectrum-dialog { width: 100% !important;} </style>`
